scio-smb/src/test/scala/org/apache/beam/sdk/extensions/smb/ParquetBucketMetadataTest.scala

```diff
@@ -253,14 +255,14 @@ class ParquetBucketMetadataTest extends AnyFlatSpec with Matchers {
       classOf[AvroGeneratedUser]
     )
 
-    metadata1.isPartitionCompatible(metadata2) shouldBe false
-    metadata2.isPartitionCompatible(metadata3) shouldBe true
-    metadata3.isPartitionCompatible(metadata4) shouldBe false
+    metadata1.isPartitionCompatibleForPrimaryKey(metadata2) shouldBe false
+    metadata2.isPartitionCompatibleForPrimaryKey(metadata3) shouldBe true
+    metadata3.isPartitionCompatibleForPrimaryKey(metadata4) shouldBe false
   }
 
   it should "check key type" in {
```

Comment:
```suggestion
  it should "parse older versions of metadata.json without a secondary key field" in {
    BucketMetadata
      .from("""
      {"type":"org.apache.beam.sdk.extensions.smb.ParquetBucketMetadata",
      "version":0,
      "numBuckets":8192,
      "numShards":1,
      "keyClass":"java.lang.String",
      "hashType":"MURMUR3_32",
      "keyField":"some_key_field"
      }""")
      .getKeyClassSecondary shouldBe null
  }

  it should "check key type" in {
```

Reply:
There are checks for version compatibility on the base `BucketMetadata` [here](https://github.com/spotify/scio/pull/4288/files#diff-2035aae344dac85e32b09945cccfab365a8ce1f9d39f19a400786a9ac6356632R137)

IIRC `getKeyClassSecondary` has to return `null` since old versions won't have anything set ... and i think being consistent with all other fields being null makes most sense.